packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -3348,193 +3351,177 @@ export default class Meeting extends StatelessWebexPlugin {
-  public cleanupLocalTracks() {
-    const {audioTrack, videoTrack, shareVideoTrack, shareAudioTrack} = this.mediaProperties;
+  public cleanupLocalStreams() {
+    const {audioStream, videoStream, shareAudioStream, shareVideoStream} = this.mediaProperties;
 
-    audioTrack?.off(LocalTrackEvents.Muted, this.localAudioTrackMuteStateHandler);
-    audioTrack?.off(LocalTrackEvents.UnderlyingTrackChange, this.underlyingLocalTrackChangeHandler);
+    audioStream?.off(StreamEventNames.MuteStateChange, this.localAudioStreamMuteStateHandler);
+    audioStream?.off(LocalStreamEventNames.OutputTrackChange, this.localOutputTrackChangeHandler);
 
-    videoTrack?.off(LocalTrackEvents.Muted, this.localVideoTrackMuteStateHandler);
-    videoTrack?.off(LocalTrackEvents.UnderlyingTrackChange, this.underlyingLocalTrackChangeHandler);
+    videoStream?.off(StreamEventNames.MuteStateChange, this.localVideoStreamMuteStateHandler);
+    videoStream?.off(LocalStreamEventNames.OutputTrackChange, this.localOutputTrackChangeHandler);
 
-    shareVideoTrack?.off(LocalTrackEvents.Ended, this.handleShareVideoTrackEnded);
-    shareVideoTrack?.off(
-      LocalTrackEvents.UnderlyingTrackChange,
-      this.underlyingLocalTrackChangeHandler
+    shareAudioStream?.off(StreamEventNames.MuteStateChange, this.handleShareAudioStreamEnded);
+    shareAudioStream?.off(
+      LocalStreamEventNames.OutputTrackChange,
+      this.localOutputTrackChangeHandler
     );
-
-    shareAudioTrack?.off(LocalTrackEvents.Ended, this.handleShareAudioTrackEnded);
-    shareAudioTrack?.off(
-      LocalTrackEvents.UnderlyingTrackChange,
-      this.underlyingLocalTrackChangeHandler
+    shareVideoStream?.off(StreamEventNames.MuteStateChange, this.handleShareVideoStreamEnded);
+    shareVideoStream?.off(
+      LocalStreamEventNames.OutputTrackChange,
+      this.localOutputTrackChangeHandler
     );
 
-    this.mediaProperties.setLocalAudioTrack(undefined);
-    this.mediaProperties.setLocalVideoTrack(undefined);
-    this.mediaProperties.setLocalShareVideoTrack(undefined);
-    this.mediaProperties.setLocalShareAudioTrack(undefined);
+    this.mediaProperties.setLocalAudioStream(undefined);
+    this.mediaProperties.setLocalVideoStream(undefined);
+    this.mediaProperties.setLocalShareAudioStream(undefined);
+    this.mediaProperties.setLocalShareVideoStream(undefined);
 
     this.mediaProperties.mediaDirection.sendAudio = false;
     this.mediaProperties.mediaDirection.sendVideo = false;
     this.mediaProperties.mediaDirection.sendShare = false;
 
-    // WCME doesn't unpublish tracks when multistream connection is closed, so we do it here
-    // (we have to do it for transcoded meetings anyway, so we might as well do for multistream too)
-    audioTrack?.setPublished(false);
-    videoTrack?.setPublished(false);
-    shareVideoTrack?.setPublished(false);
-    shareAudioTrack?.setPublished(false);
+    this.unpublishStreams([audioStream, videoStream, shareAudioStream, shareVideoStream]);
```

Comment:
this won't work, because the tracks are already unset on the mediaProperties, so unpublishStreams won't do anything

Reply:
I agree. So, we should remove the following lines from cleanUpMedia as unpublish will do it for us. Right?

```
this.mediaProperties.setLocalAudioStream(undefined);
this.mediaProperties.setLocalVideoStream(undefined);
this.mediaProperties.setLocalShareAudioStream(undefined);
this.mediaProperties.setLocalShareVideoStream(undefined);
```

Reply:
Done the same in #3053 

Kindly let me know if I have understood it incorrectly

Reply:
The changes in #3053 for this are not right. By calling unpublishStreams() you will potentially invoke code that does things like updating MuteState and releasing screen share floor, which we don't want to do. This function cleanupLocalTracks() is doing cleanup on meeting leave/destroy and just needs to clear all the references and not send any new requests to the backend. All you need to do is just put the code for cleanupLocalTracks() back to how it is on beta branch and rename all "Track" strings to "Stream". Why do you keep doing other changes?

Reply:
ah, wait, ok, there is one thing you'll have to change in cleanupLocalTracks() apart from renaming: the set of calls to `setPublished(false)` at the end, because we don't have that method anymore on the streams, so these will have to be replaced by code that emits the "meeting:streamPublishStateChanged" event with isPublished=false

Reply:
Oops. My bad. For unpublish, I misunderstood between a client unpublish and server I believe as we had this comment earlier,
> // WCME doesn't unpublish tracks when multistream connection is closed, so we do it here
> // (we have to do it for transcoded meetings anyway, so we might as well do for multistream too)

Fixed it in #3053

Reply:
> so these will have to be replaced by code that emits the "meeting:streamPublishStateChanged" event with isPublished=false

When you said so, do you mean I have to trigger an event like this?

```
Trigger.trigger(
      this,
      {
        file: 'meeting/index',
        function: 'publishStream',
      },
      EVENT_TRIGGERS.MEETING_STREAM_PUBLISH_STATE_CHANGED,
      {
        isPublished: false,
        mediaType,
        stream,
     }
);

Reply:
yes, send such event for all streams that were set (they were not undefined)

Reply:
Done in #3053 

Have written [a separate method to emit this event](https://github.com/mkesavan13/webex-js-sdk/blob/wcme3-update-importExports-variablesMethods/packages/%40webex/plugin-meetings/src/meeting/index.ts#L3497) since the code was repetitive and also updated the changes from Dipanshu to use this method.